docs/m3db/architecture/engine.md

```diff
@@ -15,95 +15,74 @@ M3DB is a persistent database with durable storage, but it is best understood vi
+3. A [`shard`](sharding.md) is effectively the same as a "virtual shard" in Cassandra in that it provides an arbitrary distribution of time series data via a simple hash of the series ID. A shard owns multiple `series`.
 
-4. `Series` which are owned by `shards`. A `series` is generally what comes to mind when you think of "time series" data. Ex. The CPU level for a single host in a datacenter over a period of time could be represented as a series with id "<HOST_IDENTIFIER>.system.cpu.utilization" and a vector of tuples in the form of (TIMESTAMP, CPU_LEVEL). In other words, if you were rendering a graph a series would represent a single line on that graph. Note that the previous example is only a logical illustration and does not represent the way that M3DB actually stores data.
+4. A `series` represents a sequence of time series datapoints. For example, the CPU utilization for a host could be represented as a series with the ID "host1.system.cpu.utilization" and a vector of (TIMESTAMP, CPU_LEVEL) tuples. Visualizing this example in a graph, there would a single line with time on the x-axis and CPU utilization on the y-axis. A `series` owns a `buffer` and multiple `block`s.
 
-5. `Blocks` belong to a series and are central to M3DB's design. A `block` is simply a smaller wrapper object around a sealed (no longer writable) stream of compressed time series data. The compression comes with a few caveats though, namely that you cannot read individual datapoints in a compressed block. In other words, in order to read a single datapoint you must decompress the entire block up to the datapoint that you're trying to read.
+5. The `buffer` is where all data that has yet to be written to disk gets stored in memory. This includes both new writes to M3DB and data obtained through bootstrapping. More details on the [buffer](engine.md#buffer) is explained below.
 
-If M3DB kept everything in memory (and in fact, early versions of it did), than you could conceptually think of it as being a composed from a hierarchy of maps:
-
-database_obect      => map<namepace_name, namespace_object>
-namespace_object    => map<shard_id, shard_object>
-shard_object        => map<series_id, series_object>
-series_object       => map<block_start_time, block_object>
-series_object       => map<block_start_time, active_buffers(encoders)> (This map should only have one or two entries)
+6. A `block` represents a stream of compressed time series data for a pre-configured block size, for example, a block could hold data for 6-8PM (block size of two hours). A `block` arrives in the series only as a result of getting cached after a read request. Since blocks are in a compressed format, individual datapoints cannot be read from it. In other words, in order to read a single datapoint, the entire block up to that datapoint needs to be decompressed beforehand.
```

Comment:
The quote "A `block` arrives in the series only as a result of getting cached after a read request" could we clarify about the fact the buffer is basically actively generating blocks?